src/types.rs

```diff
@@ -283,6 +284,28 @@ impl Data {
             },
         }
     }
+
+    /// Converts a bytestring to a canonical scalar
+    pub fn to_scalar(self) -> Result<Scalar, VMError> {
+        match self {
+            Data::Opaque(data) => {
+                let scalar = Subslice::new(&data).read_scalar()?;
```

Comment:
you can just return this without unwrapping/rewrapping the error and using a temp variable:
```suggestion
                Subslice::new(&data).read_scalar()
```